charts/redpanda/values.go

```diff
@@ -236,6 +236,7 @@ type Statefulset struct {
 	} `json:"updateStrategy" jsonschema:"required"`
 	AdditionalRedpandaCmdFlags []string          `json:"additionalRedpandaCmdFlags"`
 	Annotations                map[string]string `json:"annotations" jsonschema:"required"`
+	PodTemplateLabels          map[string]string `json:"podTemplateLabels"`
```

Comment:
Should we nest this instead to make it look more like a statefulset? IE
```go
PodTemplate struct {
    Labels map[string]string
}
```

Reply:
The annotations are used only in Pod template definition 
https://github.com/redpanda-data/helm-charts/blob/a0b1456c2274dcf85ea70f10af9b4e1a86846c1c/charts/redpanda/templates/statefulset.yaml#L52-L57

not in StatefulSet object itself
https://github.com/redpanda-data/helm-charts/blob/a0b1456c2274dcf85ea70f10af9b4e1a86846c1c/charts/redpanda/templates/statefulset.yaml#L36-L43

Reply:
Oh boy that's confusing. Hmmm. Can you toss a comment on the Annotations field so we remember that it's for the pod template? I'd still go with some nesting here, we can tackle moving/deprecating the annotations field later.